src/AuctionHouseBot.cpp

```diff
@@ -429,91 +234,52 @@ void AuctionHouseBot::addNewAuctions(Player *AHBplayer, AHBConfig *config)
             auctionEntry->buyout = buyoutPrice * stackCount;
             auctionEntry->bid = 0;
             auctionEntry->deposit = dep;
-            auctionEntry->expire_time = (time_t) etime + time(NULL);
+            auctionEntry->expire_time = elapsedTime.count() + GameTime::GetGameTime().count();
             auctionEntry->auctionHouseEntry = ahEntry;
+
             item->SaveToDB(trans);
             item->RemoveFromUpdateQueueOf(AHBplayer);
+
             sAuctionMgr->AddAItem(item);
+
             auctionHouse->AddAuction(auctionEntry);
             auctionEntry->SaveToDB(trans);
+
             CharacterDatabase.CommitTransaction(trans);
 
-            switch(itemColor)
-            {
-            case 0:
-                ++greyItems;
-                break;
-            case 1:
-                ++whiteItems;
-                break;
-            case 2:
-                ++greenItems;
-                break;
-            case 3:
-                ++blueItems;
-                break;
-            case 4:
-                ++purpleItems;
-                break;
-            case 5:
-                ++orangeItems;
-                break;
-            case 6:
-                ++yellowItems;
-                break;
-            case 7:
-                ++greyTGoods;
-                break;
-            case 8:
-                ++whiteTGoods;
-                break;
-            case 9:
-                ++greenTGoods;
-                break;
-            case 10:
-                ++blueTGoods;
-                break;
-            case 11:
-                ++purpleTGoods;
-                break;
-            case 12:
-                ++orangeTGoods;
-                break;
-            case 13:
-                ++yellowTGoods;
-                break;
-            default:
-                break;
-            }
+            itemsCount[itemColor]++;
         }
     }
 }
-void AuctionHouseBot::addNewAuctionBuyerBotBid(Player *AHBplayer, AHBConfig *config, WorldSession *session)
+
+void AuctionHouseBot::AddNewAuctionBuyerBotBid(std::shared_ptr<Player> player, std::shared_ptr<WorldSession> session, AHBConfig* config)
 {
     if (!AHBBuyer)
     {
-        if (debug_Out)
-            LOG_ERROR("module", "AHBuyer: Disabled");
+        LOG_ERROR("module.ahbot", "AHBuyer: Disabled");
         return;
     }
 
-    QueryResult result = CharacterDatabase.Query("SELECT id FROM auctionhouse WHERE itemowner<>{} AND buyguid<>{}", AHBplayerGUID, AHBplayerGUID);
+    auto sharedConfig = std::make_shared<AHBConfig>(*config);
```

Comment:
Every time we add new auctions, we copy the whole config.
The config should've been a shared_ptr to begin with..